make_profiler/parser.py

```diff
@@ -127,3 +129,47 @@ def recurse_indirect_influences(original_target, recurse_target):
             recurse_indirect_influences(original_target, t)
 
     return dependencies, influences, order_only, indirect_influences
+
+
+def check_include_instruction(filename):
+    if not os.path.isfile(filename):
+        return {}
+    
+    with open(filename, 'r') as f:
+        lines = f.read().splitlines()
+
+    # compile regex to find include instructions
+    regex = re.compile('include +')
+    # find rows which consist include instruction and replace multiple spaces with one
+    matches = [re.sub(' +', ' ', string) for string in lines if re.match(regex, string)]
+    
+    # check if input make contains include instructions
+    if len(matches) == 0:
+        temp_make_file = open(filename, 'r')
+        
+    else:
+        # create list of included makes
+        # we use join by space and then split by space to process multiple included instructions
+        makes = ' '.join([x.split('include ')[1] for x in matches]).split(' ')
```

Comment:
I believe that the order of includes is significant and includes need to happen at the spot they were mentioned.

Reply:
changed to write exactly on place of instruction with keeping order of files in instruction